### DISCUSSION TITLE
Feature Request: Amcrest Camera White Light Trigger

### DISCUSSION THREAD

**Discussion by chrisperra** [koush/scrypted#728] [Ideas]
Just an idea, spitballing it here:
I have an Amcrest camera that has the White Light on it for night vision in color, it would be super awesome if there was a way to trigger the white light either on motion or recording into HKSV. I'm playing with some settings on the Amcrest itself to see if I can have the camera do it without using the AI, but have so far had no luck.

**Comment by brianjw** [comment]
+1 - I just set up 4 new Amcrest cameras (IP8M-2779EW-AI) and they have the White Lights on them. I'd love to be able to switch this on & off via Homekit. For now, it seems only accessible via the camera GUI.

**Comment by GalagaBits** [comment]
Same here! @koush, do you think this would be possible to implement as a switch on homekit?

**Comment by jtweaver** [comment]
I also have a few IP8M-2779EB-AI cameras and they work great in scrypted. This feature is all I'm missing.

**Comment by adamkornafeld** [comment]
I am new to the NVR game, having just installed cameras over the Thanksgiving weekend. I tested cameras from Amcrest and Reolink. Saw that the Reolink camera plugin "publishes" a button on the Scrypted dashboard to control the floodlight with, whereas the Amcrest does not (yet). I opted to keep the Amcrest camera due to the better configurability as well as the fact that the Reolink _forces_ H.265 in 4K mode. 
I'll look into how that button can be added to the Amcrest plugin next. In the meanwhile triggering the functionality is fairly straightforward with code. [Here](https://gist.github.com/adamkornafeld/be7cff34e2ccdd8fda90637aea05fc24) is a gist with python and Node.js implementations.

**Reply by adamkornafeld** [reply]
Tested it with these firmwares:
> V3.000.00AC003.0.R, Build Date: 2022-08-12
> V3.000.00AC004.0.R, Build Date: 2022-10-27

**Reply by a428tm** [reply]
@adamkornafeld 
If I understood you correctly, you were able to turn on and off the LED light using the script? Is that correct?
Was this the doc you used to reference? https://support.amcrest.com/hc/en-us/articles/17903073032973-Amcrest-HTTP-API-SDK
Wanted to ask which camera model you were using, if you don't mind!

**Reply by adamkornafeld** [reply]
@a428tm, yes that's correct, the script can trigger the light on & off. The model I have is IP8M-2796EB-AI.

Thanks for the reference doc, I did not know about it. I worked backwards looking at the source code of the camera's web interface.
